src/magicbox/SuggestionsManager.ts

```diff
@@ -28,11 +30,21 @@ export enum Direction {
   Right = 'Right'
 }
 
+type KeyboardFocus =
+  | {
+      type: 'suggestion';
+      suggestion: Suggestion;
```

Comment:
Rather than having two different property names holding two different types, could we use a common interface that both the `Suggestion` class and `ISearchResultPreview` implement/extend respectively?

Also, maybe the name of the type could be more specific/descriptive. e.g.

```
export interface IQuerySuggestPreviewFocusArgs = { // just a placeholder name, feel free to shorten/change
     property1: string;
     property2: string;
}

ISearchResultPreview extends IQuerySuggestPreviewFocusArgs {}
Suggestion implements IQuerySuggestPreviewFocusArgs
```

By reducing the type to one common interface, upstream code does not have to check the `type` property of the payload. This eliminates conditionals, creates one code path, and makes the code easier to test.

Reply:
I like this, however, `Suggestion` has a property named `dom` and `ISearchResultPreview`'s equivalent property is named `element`. `element` is technically a better name but changing `dom` would affect a lot of references. Which would be better?

Reply:
One option could add a new public getter to one of the objects that returns `this.dom`. e.g.

```
// Suggestion.ts

public get element() {
    return this.dom;
}
```

Reply:
Unfortunately, `Suggestion` is an interface and transforming it into a class would also involve several changes in other files. Would making a wrapping class for `SuggestionsManager` be a good idea?

E.g.:
```ts
export class ActiveSuggestion implements IQuerySuggestPreviewFocusArgs {
    public get element() { ... }
    constructor(private suggestion: Suggestion) { ... }
}
```